repository/filtering.go

```diff
@@ -0,0 +1,78 @@
+// Package repository provides filter model
+package repository
+
+import (
+	"database/sql"
+	"log"
+
+	"github.com/google/uuid"
+)
+
+const (
+	addQuery    = "INSERT INTO FILTERS(name,type,configutation,reversed,user_id) VALUES ($1,$2,$3,$4,$5)"
+	getQuery    = "SELECT * FROM FILTERS WHERE user_id=$1"
+	updateQuery = "UPDATE FILTERS SET (type,configutation,reversed) = ($1,$2,$3) WHERE name = $4"
+	deleteQuery = "DELETE FROM FILTERS WHERE user_id=$1 AND name=$2"
+)
+
+type filtersRepo struct {
+	db *sql.DB
+}
+
+type Filters interface {
+	AllByUser(userID uuid.UUID) ([]*Filter, error)
+	Add(filter *Filter) error
+	Update(filter *Filter) error
+	Delete(userID uuid.UUID, name string) error
+}
+
+func NewFiltersRepository(db *sql.DB) Filters {

```

Comment:
The factory function should probably come first after the type declaration
In this case
- type Filters interface 
- NewFiltersRepository
- methods
Type -> Factory -> Methods -> Functions